packages/lexicon/src/serialize.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  cborDecode,
+  cborEncode,
+  check,
+  cidForCbor,
+  dataToCborBlock,
+  IpldValue,
+  ipldValueToJson,
+  jsonToIpldValue,
+  JsonValue,
+  schema,
+} from '@atproto/common'
+import { BlobRef, jsonBlobRef } from './blob-refs'
+
+export type LexValue =
+  | IpldValue
+  | BlobRef
+  | Array<LexValue>
+  | { [key: string]: LexValue }
+  | { [key: number]: LexValue }
+
+export type RepoRecord = Record<string, LexValue>
+
+export const lexValueToIpld = (val: LexValue): IpldValue => {
+  // convert blobs
+  if (val instanceof BlobRef) {
+    return val.original
+  }
```

Comment:
If we don't want to reference anything lexicon-related during serialization, we can respect `toJSON()` the same way `JSON.stringify()` itself does.  Whenever you come across an object with a `toJSON()` method during serialization, you'd replace the object with the result of `toJSON()`.  Even if it's only in the serialization path and not the deserialization path, it at least gives you an out so that repo code doesn't need to know about `BlobRef`.  `BlobRef` would just implement `toJSON()` 👍

Reply:
ah nice we probably _do_ want a `toJSON` method on this. But I think in this case, we don't want to actually return JSON, we want to return IPLD (which includes bytes & cids)